### PR TITLE
Add document:upsert

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

Add document:upsert https://github.com/kuzzleio/sdk-jvm/issues/37

### Other changes

Revert wrong change on `build.gradle.kts`
Fix mUpdate and updateByQuery doc

### CODE REVIEW

File: doc/1/controllers/document/upsert/index.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+code: true
+type: page
+title: upsert
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:4]: Missing description

File: doc/1/controllers/document/upsert/index.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+## Returns
+
+A `ConcurrentHashMap<String, Object>` with the following properties:
+
+| Property     | Type                                         | Description                      |
+|------------- |--------------------------------------------- |--------------------------------- |
+| `_source`    | <pre>ConcurrentHashMap<String, Object></pre> | Updated document (if `source` option set to true)  |
+| `_id`        | <pre>String</pre>                            | ID of the udated document                   |
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:80]:
```suggestion
| `_id`        | <pre>String</pre>                            | ID of the updated document                   |
```

File: doc/1/controllers/document/upsert/index.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+| `id`         | <pre>String</pre> | Document ID                               |
+| `changes`       | <pre>ConcurrentHashMap<String, Object></pre> | Partial content of the document to update |
+
+### Options
+
+Additional query options
+
+| Options           | Type<br/>(default)              | Description                                                                        |
+| ----------------- | ------------------------------- | ---------------------------------------------------------------------------------- |
+| `defaults` | <pre>ConcurrentHashMap<String, Object><br/>(`{}`)        | (optional) fields to add to the document if it gets created    |
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:67]:
```suggestion
| `defaults` | <pre>ConcurrentHashMap<String, Object><br/>(`{}`)        | Fields to add to the document if it gets created    |
```

File: doc/1/controllers/document/upsert/index.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+| `id`         | <pre>String</pre> | Document ID                               |
+| `changes`       | <pre>ConcurrentHashMap<String, Any?></pre> | Partial content of the document to update |
+
+### Options
+
+Additional query options
+
+| Options           | Type<br/>(default)              | Description                                                                        |
+| ----------------- | ------------------------------- | ---------------------------------------------------------------------------------- |
+| `defaults` | <pre>ConcurrentHashMap<String, Any?><br/>(`{}`)        | (optional) fields to add to the document if it gets created    |
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:116]:
```suggestion
| `defaults` | <pre>ConcurrentHashMap<String, Any?><br/>(`{}`)        | Fields to add to the document if it gets created    |
```

File: doc/1/controllers/document/upsert/index.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+| `waitForRefresh`         | <pre>Boolean</pre><br/>(`""`)    | If set to `true`, waits for the change to be reflected for `search` (up to 1s) |
+| `retryOnConflict` | <pre>Integer</pre><br/>(`10`)        | The number of times the database layer should retry in case of version conflict    |
+| `source`          | <pre>Boolean</pre><br/>(`false`)| If true, returns the updated document inside the response
+
+
+## Returns
+
+A `ConcurrentHashMap<String, Any?>` with the following properties:
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:124]:
```suggestion
A `ConcurrentHashMap<String, Any?>` object, with the following properties:
```

File: doc/1/controllers/document/upsert/index.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+| Property     | Type                                         | Description                      |
+|------------- |--------------------------------------------- |--------------------------------- |
+| `_source`    | <pre>ConcurrentHashMap<String, Any?></pre> | Updated document (if `source` option set to true)  |
+| `_id`        | <pre>String</pre>                            | ID of the udated document                   |
```
Comment (scottinet) [inline doc/1/controllers/document/upsert/index.md:129]:
```suggestion
| `_id`        | <pre>String</pre>                            | ID of the updated document                   |
```

**Review by scottinet — CHANGES_REQUESTED** [review]

**Review by Leodau — APPROVED** [review]

**Review by Shiranuit — APPROVED** [review]

**Review by Aschen — CHANGES_REQUESTED** [review]

File: doc/1/controllers/document/upsert/snippets/upsert-kotlin.kt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+val changes: ConcurrentHashMap<String, Any?> =
+  ConcurrentHashMap<String, Any?>().apply {
+    put("changes", category)  
+  }
+
+val result: ConcurrentHashMap<String, Any?> =
+  kuzzle
+  .documentController
+  .upsert("nyc-open-data", "yellow-taxi", "some-id", changes)
+  .get()
```
Comment (alexandrebouthinon) [inline doc/1/controllers/document/upsert/snippets/upsert-kotlin.kt:15]: Missing indentation

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Review by Aschen — APPROVED** [review]

**Review by scottinet — APPROVED** [review]

**Pull request merged**